tests/js/web/DropZoneInstanceManagerTest.js

```diff
@@ -0,0 +1,206 @@
+        nodeClasses: {
+            browse: 'browse',
+            info: 'info'
+        }
+    };
+
+    beforeEach(() => {
+        $html = $(document.documentElement);
+        $dropZone = $('<div class="dropzone"></div>').appendTo($html);
+        $input = $('<input type="text" id="input">').appendTo($html);
+        $browse = $('<p class="browse"></p>').appendTo($dropZone);
+        $info = $('<p class="info"></p>').appendTo($dropZone);
+
+        dropZoneStub = sinon.createStubInstance(DropZone);
+        factoryStub = sinon.stub(DropZoneFactory, 'create');
+        factoryStub.returns(dropZoneStub);
+        optionsManagerStub = sinon.createStubInstance(DropZoneOptionsManager);
+        optionsManagerStub.buildInstanceOptions.returns(stubOptions);
+        optionsManagerStub.buildValidatorOptions.returns({});
+
+        instanceManager = new DropZoneInstanceManager(
+            document.documentElement,
+            DropZoneFactory
+        );
+    });
+
+    afterEach(() => {
+        factoryStub.restore();
+        $html.html('');
+    });
+
+    describe('addInstance()', () => {
+        it('should add an instance object to memory', () => {
+            instanceManager.addInstance($dropZone[0], optionsManagerStub);
+            expect(instanceManager.instances[0]).to.deep.equal({
+                node: $dropZone[0],
+                options: stubOptions,
+                id: 0,
+                browse: $browse[0],
+                input: $input[0],
+                info: $info[0],
+                dropZone: dropZoneStub
+            });
+        });
+
+        it('should add the DropZoneInstance ID as an attribute on the node', () => {
+            instanceManager.addInstance($dropZone[0], optionsManagerStub);
+            expect($dropZone.attr('data-dropzone-id')).to.equal('0');
+        });
+
+        it('should initialise the instance', () => {
+            instanceManager.addInstance($dropZone[0], optionsManagerStub);
+            expect(dropZoneStub.init).to.have.been.calledOnce;
+        });
+    });
+
+    describe('updateInstance()', () => {
+        it('should update a property on an instance', () => {
+            instanceManager.instances = [{ id: 0, foo: 'bar' }];
+            instanceManager.updateInstance(0, 'foo', 'foo');
+            expect(instanceManager.instances[0].foo).to.equal('foo');
+        });
+    });
+
+    describe('getInstance()', () => {
+        const mockInstances = [{ id: 0 }, { id: 1 }];
+
+        it('should return all instances if no id is supplied', () => {
+            instanceManager.instances = mockInstances;
+            expect(instanceManager.getInstance()).to.deep.equal(mockInstances);
+        });
+
+        it('should return an instance from an id', () => {
+            instanceManager.instances = mockInstances;
+            expect(instanceManager.getInstance(1)).to.deep.equal(mockInstances[1]);
+        });
+    });
+
+    describe('addFiles()', () => {
+        const mockInstance = {
```

Comment:
As above, this object should probably be recreated for each test in a `beforeEach(...)`